### PR TITLE
Feature/migrating dbt project across platforms

### PR DESCRIPTION
**Pull request by shabbirk-dbtlabs**
Adds a new skill to complete dbt Fusion driven data platform migrations across dialects. Inspired by the core-to-fusion-migration skill, and the recent demos I've run at Coalesce Melbourne on the topic of various Fusion driven migration scenarios.

### Description

This was vibecoded and tested locally on my Coalesce Melbourne demo project, migrating dbt Snowflake to dbt Databricks. I will continue to test it on other scenarios and would invite other DX team members to try it too and help refine if needed.

The intended logical flow to migrate from a source data platform to a target data platform is:

- [ ] Step 1: Verify dbt Fusion is installed and working
- [ ] Step 2: Assess source project (dbtf compile — 0 errors on source)
- [ ] Step 3: Generate unit tests on source platform (Using the existing create unit test skill)
- [ ] Step 4: Switch profile to target platform
- [ ] Step 5: Run Fusion compilation and fix all errors (dbtf compile — 0 errors on target)
- [ ] Step 6: Run and validate unit tests on target platform (Fix all flagged unit test errors or identify and document data platform differences such as NULL handling, or data type differences that would require the unit test to be modified in order to pass)
- [ ] Step 7: Final validation and document changes in migration_changes.md

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-oss-template/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [ X] I have run this code in development and it appears to resolve the stated issue
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-oss-template/blob/main/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

File: skills/migrating-dbt-project-across-platforms/references/switching-profiles.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# Switching Profiles to the Target Platform
```
Comment (b-per) [inline skills/migrating-dbt-project-across-platforms/references/switching-profiles.md]: Why would we switch profiles instead of setting new targets for the same profile?

File: skills/migrating-dbt-project-across-platforms/references/installing-dbt-fusion.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+dbtf debug
+```
+
+### Install Fusion
+
+If `dbtf` is not found, install it following the official instructions:
+
+**macOS (Homebrew)**:
+```bash
+brew install dbt-labs/dbt-cli/dbtf
```
Comment (b-per) [inline skills/migrating-dbt-project-across-platforms/references/installing-dbt-fusion.md]: Does this work?

File: skills/migrating-dbt-project-across-platforms/references/installing-dbt-fusion.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+### Minimum requirements
+
+- dbt Fusion must be able to connect to both the source and target platforms
+- Run `dbtf debug` with each profile to verify connectivity before starting migration
+
+## CHALLENGES
+
+### Fusion is not available for my OS
+
+Fusion is currently available for macOS and Linux. If running on an unsupported OS, check the latest dbt Fusion documentation for updates on platform support.
```
Comment (b-per) [inline skills/migrating-dbt-project-across-platforms/references/installing-dbt-fusion.md]: Is it correct? It works on Windows as well normally (or at least WSL)

File: skills/migrating-dbt-project-across-platforms/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+#### Step 3: Generate unit tests on source platform
+
+While still connected to the **source** platform, generate dbt unit tests for key models to capture expected data outputs as a "golden dataset." These tests will prove data consistency after migration.
+
+**Which models to test**: You **must** test **every leaf node** — models at the very end of the DAG that no other model depends on via `ref()`. Do not guess leaf nodes from naming conventions — derive them programmatically using the methods in [references/generating-unit-tests.md](references/generating-unit-tests.md#identifying-leaf-nodes). List all leaf nodes explicitly and confirm the count before writing tests. Also test any mid-DAG model with significant transformation logic (joins, calculations, case statements).
+
+**How to generate tests**:
+
+1. Identify leaf nodes: `dbt ls --select "+tag:core" --resource-type model` or inspect the DAG
```
Comment (b-per) [inline skills/migrating-dbt-project-across-platforms/SKILL.md:83]: Did you want to use `dbt` here and not `dbtf`?

If so, what if people are using Fusion already but not Core?

File: skills/migrating-dbt-project-across-platforms/references/generating-unit-tests.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+dbt show --select fct_orders --limit 10
+```
+
+**Select rows that exercise key logic**:
+- Rows that hit different branches of `CASE WHEN` statements
+- Rows with NULL values in columns that have COALESCE/NVL logic
+- Rows with edge case values (zero quantities, negative amounts, boundary dates)
+- At minimum, 2-3 rows per model
+
+### Writing unit tests
```
Comment (b-per) [inline skills/migrating-dbt-project-across-platforms/references/generating-unit-tests.md:68]: Should we mention that it might good to check if the agent has access to the unit test skill before?

File: skills/migrating-dbt-project-across-platforms/references/switching-profiles.md (hunk 1 above)
Comment (shabbirk-dbtlabs) [reply skills/migrating-dbt-project-across-platforms/references/switching-profiles.md]: Ok interesting! Sorry this is me structuring my profiles.ymls like:

- profile: platform_1
----target: dev
----target: prod
- profile: platform_2
----target: dev
----target: prod

I thought this the norm so instructions suggest this. You're right we could have two targets in one profile and it'd be the same.

Is there an advantage to this?

File: skills/migrating-dbt-project-across-platforms/references/generating-unit-tests.md (hunk 5 above)
Comment (shabbirk-dbtlabs) [reply skills/migrating-dbt-project-across-platforms/references/generating-unit-tests.md:68]: Yes let's add this. Good shout

File: skills/migrating-dbt-project-across-platforms/references/switching-profiles.md (hunk 1 above)
Comment (b-per) [reply skills/migrating-dbt-project-across-platforms/references/switching-profiles.md]: Both can work technically but using target is a bit more idiomatic.

A profile is more of a 1(profile) to n(projects) relationship
- a project is tied to 1 profile (that's why it is in the `dbt_project.yml`
- 1 profile can be used by multiple projects

Again, it is more of a convention than a technical limitation

File: skills/migrating-dbt-project-across-platforms/references/installing-dbt-fusion.md (hunk 3 above)
Comment (b-per) [reply skills/migrating-dbt-project-across-platforms/references/installing-dbt-fusion.md]: FYI, it doesn't
